test/api/users_test.rb

```diff
@@ -57,4 +57,39 @@ def test_empty_users_query
 
     assert_equal [], JSON.parse(last_response.body)
   end
+
+  def test_returns_completion_for_specific_user_by_language
+    f= './test/fixtures/xapi_v3_tracks.json'
+    X::Xapi.stub(:get, [200, File.read(f)]) do
+      user = User.create(username: 'alice')
+      submission = Submission.create(user: user, language: 'Animal', slug: 'hello-world', solution: {'hello_world.rb' => 'CODE'})
+      UserExercise.create(user: user, submissions: [submission], language: 'Animal', slug: 'hello-world', iteration_count: 1)
+      submission2 = Submission.create(user: user, language: 'Fake', slug: 'apple', solution: {'apple.js' => 'CODE'})
```

Comment:
The `language` here is the track id, not the language. I really have to fix this :(